### PR TITLE
fix CentOS build errors

### PR DESCRIPTION
**Pull request by sfvaroglu**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 cmake_minimum_required(VERSION 3.0.2)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment -mf16c")
```
Comment (cconvey) [inline CMakeLists.txt]: Is this flag related to CentOS?  It seems more related to CPU architecture.

And if we're actually trying to make the best use of the CPU, would it make sense to instead use `-march=native`?

File: cmake/ngraph.cmake [hunk 2]
```diff
@@ -48,5 +48,11 @@ ExternalProject_Add(
 
 set(NGRAPH_INCLUDE_DIR ${NGRAPH_INSTALL_PREFIX}/include)
 set(NGRAPH_LIB_DIR ${NGRAPH_INSTALL_PREFIX}/lib)
+if(EXISTS "/etc/centos-release")
+    EXECUTE_PROCESS(COMMAND uname -m COMMAND tr -d '\n' OUTPUT_VARIABLE ARCHITECTURE)
+    if(${ARCHITECTURE} STREQUAL "x86_64")
+        set(NGRAPH_LIB_DIR ${NGRAPH_INSTALL_PREFIX}/lib64)
+    endif()
+endif()
```
Comment (cconvey) [inline cmake/ngraph.cmake]: CMake has a [`find_library`](https://cmake.org/cmake/help/v3.0/command/find_library.html) function.  Would it make sense to use that instead?  I think it would be a little more idiomatic, and might help us avoid having CentOS-specific logic.

File: ngraph.mk [hunk 3]
```diff
@@ -82,6 +82,21 @@ NGRAPH_BUILD_DIR := $(ROOTDIR)/3rdparty/ngraph-mxnet-bridge/build
...
+	else
+		OS := "Ubuntu"
+	endif
+endif
+
+ifeq ($(OS), "CentOS")
+	NGRAPH_LIB_DIR := $(NGRAPH_INSTALL_DIR)/lib64
+else
+	NGRAPH_LIB_DIR := $(NGRAPH_INSTALL_DIR)/lib
+endif
```
Comment (cconvey) [inline ngraph.mk]: This code assumes that any Linux computer on which this runs is either CentOS or Ubuntu.

Even if the code will happen to work okay on other major Linux's (RHEL, SuSE, etc.), it's a bit confusing to have the value of `OS` be `"Ubuntu"` for those other distributions.

Would it make sense to actually handle three cases?  I.e., Ubuntu, CentOS, and "other"?

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -19,7 +19,9 @@ get_filename_component(
     MXNET_PROJECT_SOURCE_DIR
     "${CMAKE_CURRENT_SOURCE_DIR}/../../.."
     ABSOLUTE)
-include("${MXNET_PROJECT_SOURCE_DIR}/cmake/ChooseBlas.cmake")
+if(APPLE)
+    include("${MXNET_PROJECT_SOURCE_DIR}/cmake/ChooseBlas.cmake")
+endif()
```
Comment (cconvey) [inline src/CMakeLists.txt]: Could you give a little context for this hunk?

File: CMakeLists.txt (hunk 1 above)
Comment (sfvaroglu) [reply CMakeLists.txt]: The issue shows up on CentOS, but, in general, it's for 3rdparty/mshadow definitions that we use.

File: src/CMakeLists.txt (hunk 4 above)
Comment (sfvaroglu) [reply src/CMakeLists.txt]: This was in PR #8. It causes blas problems on CentOS. cmake complains about not finding Atlas, even though Atlas is already installed and `USE_BLAS` is not used, e.g. `make -j$(nproc) USE_NGRAPH=1` fails to build due to Atlas problems. Is ChooseBlas needed for OS X?

Comment (cconvey) [reply src/CMakeLists.txt]: > Is ChooseBlas needed for OS X?

I'm not sure, but I would suggest testing this on OS X before merging the PR.

If you decide to keep this hunk, it might be nice to add a small comment explaining what you told me.

Comment (cconvey) [reply src/CMakeLists.txt]: Here's my concern.  There are two bodies of code that use the mshadow headers: 
- The mxnet-ngraph bridge code
- The rest of mxnet

The reason we include `ChoseBlas.cmake` here is to ensure that when both of those bodies of code are compiled, the same preprocessor flags are in effect during the compilation of the mshadow headers.  It's not clear what the consequences of a mismatch would be on Linux, but IIRC on OSX it was problematic.

So all things considered, I'd be more comfortable figuring out:
- what's going wrong with this logic on CentOS, and
- why the rest of MXnet isn't broken on CentOS due to `ChoseBlas.cmake`.

That being said, there's an outstanding upstream PR for refactoring / rationalizing how MXnet's CMake build system deals with math libraries.  So if this passes testing I think we can live with it as an interim solution.
